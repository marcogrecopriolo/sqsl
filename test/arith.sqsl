{ ints }

{ precedence }
display "1+2*3";
display 1+2*3 format full;

display "8/4-1";
display 8/4-1 format full;

{ grouping }
display "(1+2)*3";
display (1+2)*3 format full;

display "9/(4-1)";
display 9/(4-1) format full;

{ floats }

{ precedence }
display "1.0+2.1*3.2";
display 1.0+2.1*3.2 format full;

display "8.2/4.1-1.0";
display 8.2/4.1-1.0 format full;

{ grouping }
display "(1.0+2.1)*3.2";
display (1.0+2.1)*3.2 format full;

display "9.2/(4.1-1.0)";
display 9.2/(4.1-1.0) format full;

{ expression type }
display "1+2.1";
display 1+2.1 format full;

display "8.2/2";
display 8.2/2 format full;

{ nulls }
display "1+null";
display 1+null format full;

{ casting }
display "1.1::int";
display 1.1::int format full;

{ booleans }
display "1>2";
display 1>2 format full;

display "1<2";
display 1<2 format full;
