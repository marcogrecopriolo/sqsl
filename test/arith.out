1+2*3
7
8/4-1
1
(1+2)*3
9
9/(4-1)
3
1.0+2.1*3.2
7.72
8.2/4.1-1.0
1
(1.0+2.1)*3.2
9.92
9.2/(4.1-1.0)
2.96774
1+2.1
3.1
8.2/2
4.1
1+null

1.1::int
1
1>2
0
1<2
1
